bam_sort.c

```diff
@@ -1664,7 +1664,7 @@ static inline int bam1_lt_core(const bam1_p a, const bam1_p b)
 uint8_t normalize_type(const uint8_t* aux) {
     if (*aux == 'c' || *aux == 'C' || *aux == 's' || *aux == 'S' || *aux == 'i' || *aux == 'I') {
         return 'c';
-    } else if (*aux == 'f' || *aux == 'F') {
+    } else if (*aux == 'f' || *aux == 'd') {
```

Comment:
I don't think so.  [The sam specification](http://samtools.github.io/hts-specs/SAMv1.pdf) lists `AcCsSiIfZHB` as valid type specifiers on page 12.  `d` is a historic samtools/htslib extension for "double" - I don't think it's ever written, but support for reading it is there.